classes/village/VillageProposalDto.php

```diff
@@ -14,6 +14,7 @@ public function __construct(
         public ?int $policy_id,
         public ?string $vote_time_remaining,
         public ?string $enact_time_remaining,
+        public array $trade_data = [],
```

Comment:
I think this should really be its own DTO, it's not a variable-length list of things, it's used as a structured data object. Especially as there are multiple levels of nested, labeled fields here and it's hard to track down all of them

Reply:
Yeah that makes sense, while developing it was easier to just grab what I needed while experimenting but especially as this grows to more complex trades, nested arrays will get even more messy. The next time we touch trading (e.g. adding the ability to create counter-offers) I'll rework this to a DTO.